api/controllers/Irc_EntryController.js

```diff
@@ -117,31 +119,97 @@ module.exports = {
     switch (request_body.operation) {
     case Event.OPERATION_INTER_SITE_TRANSFER:
       return processEventDetainee(request_body)
-          .then((detainee) => this.handleInterSiteTransfer(detainee, request_body));
+        .then((detainee) => this.handleInterSiteTransfer(detainee, request_body));
     case Event.OPERATION_UPDATE:
       return processEventDetainee(request_body)
-          .tap((detainee) => Centres.publishUpdateOne(detainee.centre));
+        .tap((detainee) => Centres.publishUpdateOne(detainee.centre));
     case Event.OPERATION_CHECK_IN:
     case Event.OPERATION_CHECK_OUT:
     case Event.OPERATION_REINSTATEMENT:
       return processEventDetainee(request_body)
-          .then((detainee) => Event.create(generateStandardEvent(detainee, request_body)))
-          .tap((event) => Centres.publishUpdateOne(event.centre));
+        .then((detainee) => Event.create(generateStandardEvent(detainee, request_body)))
+        .tap((event) => Centres.publishUpdateOne(event.centre));
+    case BedEvent.OPERATION_OUT_OF_COMMISSION:
+    case BedEvent.OPERATION_IN_COMMISSION:
+      return this.processBedEvent(request_body)
+        .tap((event) => Centres.publishUpdateOne(event.centre_id));
     default:
       throw new ValidationError('Unknown operation');
     }
   },
 
+  findAndPopulateCentre: event =>
+    Centres.getByName(event.centre)
+      .then((result) => _.merge(event, {centre_id: result.id})),
+
+  findAndPopulateDetainee: (event) => {
+    let detainee;
+    if (_.isNull(event.single_occupancy_person_id) || _.isUndefined(event.single_occupancy_person_id)) {
+      detainee = Promise.resolve(_.merge(event, {detainee: null}));
+    } else {
+      detainee = createOrUpdateDetainee({
+        person_id: event.single_occupancy_person_id,
+        centre: {id: event.centre_id},
+        timestamp: new Date().toString()
+      })
+        .then((result) =>
+          _.merge(event, {detainee: result.id}));
```

Comment:
`_.merge` returns the object whereas setting a property on the object directly returns the value of the property.